src/components/client/data/RawData.tsx

```diff
@@ -14,33 +21,52 @@ export const RawData: React.FC = () => {
 
   const toast = useToast();
 
-  const onToggle = () => {
```

Comment:
encoding here is the radio button that the user is switcing to. so it should be if encoding is hex, we decode the bs58 and encode to hex.